src/complex/Filter/FilterList.cpp

```diff
@@ -45,12 +62,24 @@ AbstractPlugin* FilterList::getPluginById(const FilterHandle::PluginIdType& id)
   return nullptr;
 }
 
+IFilter::UniquePointer FilterList::createFilter(const std::string& humanOrClassName) const
+{
+  for(const auto& handle : getFilterHandles())
+  {
+    if(handle.getFilterName() == humanOrClassName || handle.getClassName() == humanOrClassName)
+    {
+      return createFilter(handle);
+    }
+  }
+  return nullptr;
+}
```

Comment:
I believe it is fundamentally a bad design to use only the class or human name. Having two perfectly valid plugins should not destroy the ability to instantiate a filter for no other reason than they both declared a filter with the same name.

If it was impossible for anyone outside of BlueQuartz to create plugins at any point, it would purely be our fault if this method failed. If we added a filter that shared a name with a filter another organization developed for their own internal workflow, when, if ever, would a user be notified that the incorrect filter was being instantiate? Would their plugin be allowed to be imported at all? Is it their job to rename their classes whenever we release new filters? What if that is not in their budget? What if a user / developer had pre-compiled plugins from multiple organizations with a naming conflict they couldn't change?

Do NOT use names. There are too many ways it can fail miserably. If you have to, use the search method and force the caller to pick a FilterHandle if multiple filters were found. FilterHandles were made to be the sole way to instantiate filters for this reason.